### ISSUE TITLE
_getVerifiedStorageObject - contentious sessionStorage element

### ISSUE THREAD

**Issue by arfraz** [microsoft/ApplicationInsights-JS#2094]
See:  function _getVerifiedStorageObject  within: 

https://github.com/microsoft/ApplicationInsights-JS/blob/main/shared/AppInsightsCommon/src/StorageHelperFuncs.ts

This function sets the name of the element to the same as value being a (new Date).toString();
  storage.setItem(uid, uid);

Online scanning tool Facette has reported this is violating GDPR.   The name of the element needs to be set to something appropriate as to ensure the element can be deemed as 'necessary'

In other words, with the date time being stored as the name of the element, it cannot be identified as a necessary element.

can we set the name of the element to be something static like:  sessionStorageCheck  so that we mark this value as necessary.

**Comment by MSNev** [comment]
Because we support multiple instances on a page and there may be multiple instances running across multiple tabs, we need the value to be unique.....

We could prefix the value, would that suffice? So that you can identify that <prefix>* are necessary?

**Comment by arfraz** [comment]
thanks for the feedback.   As an alternative could I suggest we capture all of them for multiples instances within an array of objects thereby allowing the name of the element to be static ?

You can then run the check by parsing the stringify'd array of objects and fetching the datestring (uid variable) within it.

**Comment by MSNev** [comment]
>  allowing the name of the element to be static

This won't really work when they are coming from different tabs (which are completely disconnected from a runtime perspective) and also when we have multiple components for multiple "teams" reporting with different iKeys.

**Comment by arfraz** [comment]
thanks for checking that.  At the same time I am reviewing whether a prefix is sufficient enough to comply and will let you know.

**Comment by arfraz** [comment]
have received some feedback that the approach you mentioned with the prefix and wildcard will work.   If you're able to implement that and let me know when ready, that would be great.  

thanks,
Andrew

**Comment by MSNev** [comment]
I've created an internal work item for this, not sure of the exact timeframe at this point.

**Comment by MSNev** [comment]
Now fully released as part of version 2.8.15

**Comment by MSNev** [comment]
Linking to PR's v2.x #2114, v3.x #2110

**Comment by arfraz** [comment]
Hi @MSNev ,   what is the prefix supposed to be?  We've been testing out the code change but not seeing the prefix applied.   What should we be seeing.   With the new version we are still getting the following as per attached screenshot.

![MicrosoftTeams-image](https://github.com/microsoft/ApplicationInsights-JS/assets/103406920/cddc0e11-4196-4b9a-a1b0-832f07b9479e)

**Comment by MSNev** [comment]
If you want a specific prefix for the "standard" values then you should use the previous existing `namePrefix` configuration value.

The new value added is ONLY for the `_getVerifiedStorageObject` when it creates the "temporary" item to verify that it can access the requested storage type (Session or LocalStorage), it's created and removed almost instantly.
